pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -78,21 +78,25 @@ func NewObjectMappingFactory(clientAccessFactory ClientAccessFactory) ObjectMapp
 // the built in mapper if necessary. It supports unstructured objects either way, since
 // the underlying Scheme supports Unstructured. The mapper will return converters that can
 // convert versioned types to unstructured and back.
+// It can return an error and the best effort unstructured mapper and typer.
 func (f *ring1Factory) objectLoader() (meta.RESTMapper, runtime.ObjectTyper, error) {
 	discoveryClient, err := f.clientAccessFactory.DiscoveryClient()
 	if err != nil {
-		glog.V(3).Infof("Unable to get a discovery client to find server resources, falling back to hardcoded types: %v", err)
-		return legacyscheme.Registry.RESTMapper(), legacyscheme.Scheme, nil
+		unstructuredMapper := discovery.NewRESTMapper(nil, meta.InterfacesForUnstructured)
```

Comment:
How does this look like in kubectl? The error is printed as a warning?

Reply:
> How does this look like in kubectl? The error is printed as a warning?

The only way we've managed to do it is with an invalid user which also fails when building any other client, so this just fails it a little earlier.